commons/src/main/java/io/aiven/kafka/tieredstorage/commons/storage/filesystem/FileSystemStorage.java

```diff
@@ -73,12 +74,21 @@ public InputStream fetch(final String key, final BytesRange range) throws Storag
         try {
             final Path path = fsRoot.resolve(key);
             final long fileSize = Files.size(path);
-            if (range.to > fileSize) {
-                throw new IllegalArgumentException("position 'to' cannot be higher than the file size, to="
-                    + range.to + ", file size=" + fileSize + " given");
+            if (fileSize > Integer.MAX_VALUE) {
+                throw new IOException("File size is larger than " + Integer.MAX_VALUE);
```

Comment:
Do you think it should be an `IOException`, not `IllegalArgumentException` or `InvalidRangeException`?

Reply:
`fileSize` is not an input (to use IllegalArgExc), but a size larger than int bounds that may potentially break the casting, since nio.Files#size now return long instead of int. 
Also, not range to use the new InvalidRangeException.
So, IOException seems the closest exception, but there may be a better type?

Reply:
This IMO is in the wrong place. We are requesting the range and before we check that `to` is bigger than `fileSize` we should not care too much I guess since even if file is too big we can still read a part of it.

Reply:
@AnatolyPopov not really, as we are using the filesize to set the bounded inputstream later. The ranges expected by the framework as not considering files larger than max int--I guess as https://kafka.apache.org/documentation/#brokerconfigs_log.segment.bytes is defined as int-- so it may be worth to do this check here.

Reply:
We are not using fileSize to set bounds for BoundedImputStream directly. I'm trying to say that this check should check if `Math.min(range.to + 1, (int) fileSize)` is less than `Integer.Max_VALUE`. Moreover BoundedInoutStream relies on the size of underlying input stream and unless the size of BoundedInputStream is smaller than the size of underlying stream it does not matter how big it is.

Reply:
I see your point, refactoring this validation. Thanks @AnatolyPopov!